src/app/app.component.ts

```diff
@@ -307,7 +307,7 @@ export class AppComponent implements OnInit, AfterViewInit {
         window['startUpload'] = (selectFile, bucketId, uploadOptions,folderId, cb) => {
             window['isUpload'] = true;
             this.showPrompt =  true;
-            if(folderId !=""){
+            if(folderId !== null && folderId !== ""){
```

Comment:
Why to check not equal type?

Reply:
It is a check whether the current location is inside a folder. If we create a folder then the folderId will have the folder name.
We check whether the folderId is not empty and not null.
On fresh installation of SODA, the variable that holds the folder name is not set and is equal to null. This was causing the object name to be prefixed with null as there was only a check on whether the folder name was not empty. null is not empty although they both solve to false.

Reply:
On re-reading this I realise you were asking about the type check. 

The issue we have is that on first load of SODA Dashboard when we have not created a folder the object folderId is not initiated. When the value is checked it shows up as null. As mentioned in the second link below if it is undefined the type could be undefined and if it is null the type is object. 
Hence the type check.

More info [here](https://medium.com/javascript-in-plain-english/how-to-check-for-null-in-javascript-dffab64d8ed5#:~:text=Comparisons%20can%20be%20made%3A%20null,a%20value%20is%20not%20null%20.) and [here](https://stackoverflow.com/a/16204960/6303009)

Reply:
Ok

Reply:
Good so now the value and type both are checked

Reply:
Yes. that is correct.